### PR TITLE
User recommendations, level 1

### PR DESCRIPTION
**Pull request by brianloveswords**
Right now we are doing the following:
- Filter earned badges
- Filter category-level badges
- Filter offline badges

We store this collection as `allBadges`, then
- Filter participation badges
- Filter badges that belong to categories where the user has already
  earned the category-level badge
- Filter badges that are "off-track", meaning, badges in categories
  where the user hasn't completed any other badges.

And we store this collection as `filtered`. If `filtered` is an empty
set, we shuffle `allBadges` and recommend that set.

There are a number of ways this can be improved:
- Only recommend age appropriate badges.
- Deal with offline badges: right now we are not recommending them
  because getting access to the program start date is annoying, and we
  don't want to recommend badges that haven't started yet
- Be smarter about recommending badges that will complete a category
  level badge.
- Be smarter about falling back when a filter reduces the recommendation
  set to zero items. For example, in the case where a user hasn't earned
  any badges yet, we are going to fall back completely to `allBadges`,
  but we should probably still filter out participation badges.

### CODE REVIEW

**Comment by cmcavoy** [comment]
Keeping offline badges out of the recommendations is kind of rough...they're like 90% of the badges in the system. /cc @erinknight

**Comment by brianloveswords** [comment]
You said to do it in stages, this is stage 1. Offline badges are coming back in later stages.

**Comment by erinknight** [comment]
I'm okay with only doing online badges for stage 1. There are enough online badges to get people to the city level badges so we know people can get there.

**Comment by cmcavoy** [comment]
Ok :+1: all around...will merge and deploy momentarily.

File: tests/badge-recommendations.test.js [hunk 1]
```diff
@@ -20,13 +59,28 @@ test.applyFixtures({
     categories: ['math'],
     name: 'Math',
     description: 'for mathing',
+    type: 'skill',
+    activityType: 'online',
+    categoryAward: '',
+    image: Buffer(1),
+  }),
+  'pure-knitting': new Badge({
```
Comment (cmcavoy) [inline tests/badge-recommendations.test.js:67]: heheh.

**Pull request merged**